packages/flutter_tools/lib/src/http_host_availability_validator.dart

```diff
@@ -0,0 +1,142 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.9
+
+import 'base/io.dart';
+import 'base/platform.dart';
+import 'doctor.dart';
+
+// The environment variables used to override some URLs
+const String kPubHostedUrl = 'PUB_HOSTED_URL';
+const String kCloudUrl = 'FLUTTER_STORAGE_BASE_URL';
+
+/// Validation class that checks if all the given URLs are reachable
+class HttpHostAvailabilityValidator extends DoctorValidator {
+  HttpHostAvailabilityValidator({
+    Platform platform,
+    HttpClient httpClient,
+  })   : _platform = platform,
+        _httpClient = httpClient,
+        super('HTTP host availability');
+
+  final Platform _platform;
+  final HttpClient _httpClient;
+
+  @override
+  String get slowWarning => 'HTTP host availability check is taking a long time';
+
+  /// Returns a list of URLs to check availability, different for different platforms
+  List<String> get _allRequiredHosts {
+    /// Hosts used by flutter on all machines
+    final List<String> commonRequiredHostUrls = <String>[
```

Comment:
One more adjustment I would make - use the `Features` class to determine if android support is enabled, and do not check the host if it isn't. Same thing for iOS:

https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/features.dart#L43

Reply:
Should I check if the host platform is MacOS now that the validator is checking for the IOS feature being enabled?

Reply:
Done, thanks.